lib/block_view.go

```diff
@@ -3496,17 +3507,88 @@ func (bav *UtxoView) ValidateDiamondsAndGetNumDeSoNanos(
 }
 
 func (bav *UtxoView) ConnectTransaction(
-	txn *MsgDeSoTxn, txHash *BlockHash,
-	blockHeight uint32, blockTimestampNanoSecs int64, verifySignatures bool,
-	ignoreUtxos bool) (_utxoOps []*UtxoOperation, _totalInput uint64, _totalOutput uint64, _fees uint64, _err error) {
-	return bav._connectTransaction(txn, txHash, blockHeight, blockTimestampNanoSecs, verifySignatures, ignoreUtxos)
-
+	txn *MsgDeSoTxn,
+	txHash *BlockHash,
+	blockHeight uint32,
+	blockTimestampNanoSecs int64,
+	verifySignatures bool,
+	ignoreUtxos bool,
+) (
+	_utxoOps []*UtxoOperation,
+	_totalInput uint64,
+	_totalOutput uint64,
+	_fees uint64,
+	_err error,
+) {
+	return bav._connectTransaction(
+		txn,
+		txHash,
+		blockHeight,
+		blockTimestampNanoSecs,
+		verifySignatures,
+		ignoreUtxos,
+	)
 }
 
 func (bav *UtxoView) _connectTransaction(
-	txn *MsgDeSoTxn, txHash *BlockHash, blockHeight uint32,
-	blockTimestampNanoSecs int64, verifySignatures bool,
-	ignoreUtxos bool) (_utxoOps []*UtxoOperation, _totalInput uint64, _totalOutput uint64, _fees uint64, _err error) {
+	txn *MsgDeSoTxn,
+	txHash *BlockHash,
+	blockHeight uint32,
+	blockTimestampNanoSecs int64,
+	verifySignatures bool,
+	ignoreUtxos bool,
+) (
+	_utxoOps []*UtxoOperation,
+	_totalInput uint64,
+	_totalOutput uint64,
+	_fees uint64,
+	_err error,
+) {
+	// If the transaction is actually a series of atomic transactions, we process the transaction via
+	// _connectAtomicTransactionsWrapper which will recursively call each inner transaction as
+	// well as provide cumulative fee checking for the atomic transactions.
+	if txn.TxnMeta.GetTxnType() == TxnTypeAtomicTxnsWrapper {
+		return bav._connectAtomicTxnsWrapper(
+			txn,
+			txHash,
+			blockHeight,
+			blockTimestampNanoSecs,
+			verifySignatures,
+			ignoreUtxos,
+		)
+	}
+
+	// Check that we're not trying to commit a transaction meant to be part of a series of atomic transactions
+	// outside an atomic transactions wrapper.
+	if txn.IsAtomicTxnsInnerTxn() {
+		return nil, 0, 0, 0, RuleErrorAtomicTxnsRequiresWrapper
+	}
+
+	// By here, we should know the transaction to be non-atomic.
+	return bav._connectNonAtomicTransaction(
```

Comment:
I feel like we should have a better name for this. Right now you actually do use it to connect an atomic txn later on so it's not a great name. Maybe like _connectSingleTxn? Idk noodle on it.

Reply:
I like `_connectSingleTxn`